test/api/clustering_test.cpp

```diff
@@ -1,19 +1,319 @@
+        }
+    };
+
+    ASSERT_EQ(expected_partitions.size(), partitions.size());
+
+    for (size_t partition_index = 0; partition_index < expected_partitions.size(); ++partition_index)
+    {
+        ASSERT_EQ(expected_partitions[partition_index].size(), partitions[partition_index].size());
+        for (size_t junction_index = 0; junction_index < expected_partitions[partition_index].size(); ++junction_index)
+        {
+            EXPECT_TRUE(expected_partitions[partition_index][junction_index] == partitions[partition_index][junction_index]);
+        }
+    }
+}
+
+
+TEST(clustering, strict_clustering)
+{
+    std::vector<Junction> input_junctions = prepare_input_junctions();
+    std::vector<Cluster> clusters = hierarchical_clustering_method(input_junctions, 0);
+
+    // Each junction in separate cluster
+    std::vector<Cluster> expected_clusters
+    {
+        Cluster{{   Junction{Breakend{chrom1, chrom1_position1 - 5, strand::forward},
+                             Breakend{chrom2, chrom2_position1 + 8, strand::forward}, ""_dna5, read_name_1}
+        }},
+        Cluster{{   Junction{Breakend{chrom1, chrom1_position1 + 2, strand::forward},
+                             Breakend{chrom2, chrom2_position1 - 3, strand::forward}, ""_dna5, read_name_2}
+        }},
+        Cluster{{   Junction{Breakend{chrom1, chrom1_position1 + 9, strand::forward},
+                             Breakend{chrom2, chrom2_position1 + 1, strand::forward}, ""_dna5, read_name_3},
+        }},
+        Cluster{{   Junction{Breakend{chrom1, chrom1_position1 + 5, strand::forward},
+                             Breakend{chrom2, chrom2_position1 - 1, strand::reverse}, ""_dna5, read_name_4},
+        }},
+        Cluster{{   Junction{Breakend{chrom1, chrom1_position1 + 92, strand::forward},
+                             Breakend{chrom2, chrom2_position1 + 3, strand::forward}, ""_dna5, read_name_5},
+        }},
+        Cluster{{   Junction{Breakend{chrom1, chrom1_position2 - 2, strand::forward},
+                             Breakend{chrom2, chrom1_position3 + 8, strand::reverse}, ""_dna5, read_name_6}
+        }},
+        Cluster{{   Junction{Breakend{chrom1, chrom1_position2 + 3, strand::forward},
+                             Breakend{chrom2, chrom1_position3 - 1, strand::reverse}, ""_dna5, read_name_7}
+        }},
+        Cluster{{   Junction{Breakend{chrom1, chrom1_position2 + 6, strand::forward},
+                             Breakend{chrom2, chrom1_position3 + 2, strand::reverse}, ""_dna5, read_name_8}
+        }}
+    };
+    std::sort(expected_clusters.begin(), expected_clusters.end());
+
+    ASSERT_EQ(expected_clusters.size(), clusters.size());
+
+    for (size_t cluster_index = 0; cluster_index < expected_clusters.size(); ++cluster_index)
+    {
+        ASSERT_EQ(expected_clusters[cluster_index].get_cluster_size(), clusters[cluster_index].get_cluster_size()) << "Cluster " << cluster_index << " of unexpected size";
```

Comment:
is it possible to split this line?
By the way, I don't know this `ASSERT(...) << bla` at all, that you can write an output behind it. You never stop learning. :)

Reply:
https://github.com/google/googletest/blob/master/docs/primer.md#assertions